project/dexter-jenkins/src/main/java/io/jenkins/plugins/dexter/DexterPublisher.java

```diff
@@ -238,27 +244,38 @@ public boolean perform(AbstractBuild build, Launcher launcher, BuildListener lis
 				} else
 					message = "Error saving logs";
 			} else {
+
 				writeToShFile();
-				Runtime runtime = Runtime.getRuntime();
-				String command = "/home/v.sanko/dexter-server/run.sh";
-				try {
-					runtime.exec(command);
-				} catch (IOException e) {
-					e.printStackTrace();
-				}
 				writeToFile();
 
-				String commandDexter = pathToBat + "/dexter.sh " + dexterUser + " " + dexterPassword + " > logs.txt \"";
 				try {
-					runtime.exec(commandDexter);
-				} catch (IOException e) {
+					message = "";
+					int ch;
+
+					ProcessBuilder pb = new ProcessBuilder(dexterServerPath + "/run.sh");
+					pb.directory(new File(dexterServerPath));
+
+					Process shellProcess = pb.start();
+
+					ProcessBuilder pbClient = new ProcessBuilder(pathToBat + "/dexter.sh", dexterUser, dexterPassword);
+					pbClient.directory(new File(pathToBat));
+
+					Process shellProcessClient = pbClient.start();
+
+					InputStreamReader myIStreamReader = new InputStreamReader(shellProcessClient.getInputStream());
+					
+					while ((ch = myIStreamReader.read()) != -1) {
```

Comment:
Does it have to be read char-by-char, instead of e.g. reading whole String at once?

Reply:
As I answer you before, I tried to do this with the help of BufferedReader(myIStreamReader).lines().collect(Collectors.joining(System.lineSeparator())) but unfortunately build fails because of findbugs error (no encoding specified). I can try do define the charset (UTF-8) or just disable findbugs during build, or leave this low level but working code, as you want. What is better solution?

Reply:
IMO defining the charset would be the best solution.

Reply:
fixed